tracing/trace_state.cc

```diff
@@ -54,6 +54,13 @@ namespace tracing {
 
 logging::logger trace_state_logger("trace_state");
 
+struct statement_param {
+    // provides an access to the cql3 statements of present prepared statements
+    shared_ptr<cql3::cql_statement> cql;
```

Comment:
Is this for `raw_cql_statement`? Maybe we should make it available in `prepared_statement`. In fact I see that prepared_statement has a reference to cql_statement. So I don't understand this.

Reply:
I think the chances of losing the prepared statement are practically nil.

Reply:
> I think the chances of losing the prepared statement are practically nil.

It's not zero and it depends on a configuration controlled by the user (it depends on the size of the cache).
And we don't need to assume anything after this patch.
This patch replaces copying the string to holding the shared_ptr. 

As a result the prepared_statement can be safely evicted and would still have a query string without having to copy it until we really need it.

Reply:
reworked to use checked weak ptr with: `statements[i] ? statements[i]->statement->raw_cql_statement : "<missing>"`